dir2cast.php

```diff
@@ -1881,7 +1881,9 @@ public static function defaults(array $SERVER)
 
         if(!defined('TITLE'))
         {
-            if(basename(MP3_DIR()))
+            if(file_exists(MP3_DIR() . 'title.txt'))
+                define('TITLE', file_get_contents(MP3_DIR() . 'title.txt'));
+            elseif(basename(MP3_DIR()))
```

Comment:
Ok! So this change looks legit to me, and although you didn't explicitly say this, it makes sense for the configuration where one deployment of dir2cast is powering multiple sun folders (otherwise you could achieve the same thing just by editing dir2cast.ini and setting the title there).

Two things spring to mind:
 1. From the description you gave of the problem you are trying to solve, it occurs to me that perhaps MP3_DIR() should be doing a urldecode() somewhere so that the %20 space characters are not in displayed in the title. Please let me know if that is actually the problem! Regardless of that, I can see other uses (titles with characters that you just aren't allowed to have in a folder name such as '/')
 2. You didn't add a unit test for the new functionality. I realise that adding those is generally quite boring 😂 so I'm happy to add one for you unless you beat me to it.

Reply:
>1. From the description you gave of the problem you are trying to solve, it occurs to me that perhaps MP3_DIR() should be doing a urldecode() somewhere so that the %20 space characters are not displayed in the title. Please let me know if that is actually the problem! Regardless of that, I can see other uses (titles with characters that you just aren't allowed to have in a folder name such as '/')

Correct! So the exact use case for me is that I'm using dir2cast in a Docker container to host my own audiobooks as a Podcast. This allows me to add the audiobook as an RSS feed to [Snipd](https://www.snipd.com/), which has the ability to 'highlight' parts of the audio. Every audiobook therefore gets it's own folder. However, Snipd doesn't seem to handle image URLs with spaces very well. Reminder to myself to log that as a defect there as well :).

>2. You didn't add a unit test for the new functionality. I realise that adding those is generally quite boring 😂 so I'm happy to add one for you unless you beat me to it.

Ah sorry, missed that one! I'm happy to add a Unit test myself. However, I have no experience in PHP (I made the edit based on copying what you did already haha). So if you could point towards some documentation on how you've structured these tests, then I'll try to see if I can build something that works :).

Reply:
> Ah sorry, missed that one! I'm happy to add a Unit test myself. However, I have no experience in PHP (I made the edit based on copying what you did already haha). So if you could point towards some documentation on how you've structured these tests, then I'll try to see if I can build something that works :).

Really no ptoblem. If you look for the tests which reference `description.txt` (or other files which are on adjacent lines to your change on line 2120) and tests which reference `TITLE` this may be enough to get going - but it's probably not the most straightforward. If you can't make head nor tail of it, let me know and I'll do it.